pkg/compose/publish.go

```diff
@@ -18,24 +18,138 @@ package compose
 
 import (
 	"context"
+	"encoding/json"
+	"os"
 
 	"github.com/compose-spec/compose-go/types"
 	"github.com/distribution/reference"
+	"github.com/docker/buildx/util/imagetools"
 	"github.com/docker/compose/v2/pkg/api"
+	"github.com/docker/compose/v2/pkg/progress"
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/specs-go"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 func (s *composeService) Publish(ctx context.Context, project *types.Project, repository string, options api.PublishOptions) error {
+	return progress.RunWithTitle(ctx, func(ctx context.Context) error {
+		return s.publish(ctx, project, repository, options)
+	}, s.stdinfo(), "Publishing")
+}
+
+func (s *composeService) publish(ctx context.Context, project *types.Project, repository string, options api.PublishOptions) error {
 	err := s.Push(ctx, project, api.PushOptions{})
 	if err != nil {
 		return err
 	}
 
-	_, err = reference.ParseDockerRef(repository)
+	w := progress.ContextWriter(ctx)
+
+	named, err := reference.ParseDockerRef(repository)
+	if err != nil {
+		return err
+	}
+
+	resolver := imagetools.New(imagetools.Opt{
+		Auth: s.configFile(),
+	})
+
+	var layers []v1.Descriptor
+	for _, file := range project.ComposeFiles {
+		f, err := os.ReadFile(file)
+		if err != nil {
+			return err
+		}
+
+		w.Event(progress.Event{
+			ID:     file,
+			Text:   "publishing",
+			Status: progress.Working,
+		})
+		layer := v1.Descriptor{
+			MediaType: v1.MediaTypeImageLayer,
+			Digest:    digest.FromString(string(f)),
```

Comment:
```suggestion
			Digest:    digest.FromBytes(f),
```

Internally, `digest.FromString` will just convert back to bytes so might as well pass it directly